### PR TITLE
Fix a bug with extract function parameter calculation

### PR DESCRIPTION
**Pull request by jmbockhorst**
This fixes a case where the extract function code action would incorrectly add parameters for inner destructured record fields. Also adds tests for other cases.

### CODE REVIEW

File: test/utils/sourceParser.ts [hunk 1]
```diff
@@ -159,27 +157,35 @@ export function getInvokeAndTargetPositionFromSource(source: string): TestType {
 
 export function getTargetPositionFromSource(
   source: string,
-): { position: Position; sources: { [K: string]: string } } | undefined {
+): { range: Range; sources: { [K: string]: string } } | undefined {
```
Comment (razzeee) [inline test/utils/sourceParser.ts:160]: Hrm, not your fault, but I think we don't use range at all here?

Comment (razzeee) [reply test/utils/sourceParser.ts:160]: Ignore me, those are the return values :D

File: test/codeActionTests/extractFunctionCodeAction.test.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+foo val str =
+    case val of
+        Just { field1, field2, field3 } ->
+            case field1 of
+          --^
+                Just { prop1, prop2 } ->
+                    prop1 + prop2 + val + field2
+
+                Nothing ->
+                               --^
```
Comment (razzeee) [inline test/codeActionTests/extractFunctionCodeAction.test.ts:21]: Are these supposed to be off by a line? I would expect them to be moved down by one?

Comment (jmbockhorst) [reply test/codeActionTests/extractFunctionCodeAction.test.ts:21]: Yeah, originally I had that, but only this worked. At the time I didn't dig into why, but I will now.

**Pull request merged**